### PR TITLE
render list items on List component

### LINKED ISSUE #1: 1. As a user, I want to read the names of all of the items in my shopping list
## Summary

The most important feature of our app is the ability to look at what's on the list, so the user knows what they need to buy. The `List` component is the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `streamListItems` function is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` to show more information – you will do that **later**. If you want a peek at the future, [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png) is what your list could look like closer to the end of the project.
- The data you need is already being passed into the `List` component. What steps do you need to take in order to render it?

### PR DESCRIPTION
**Pull request by Amy-Pr**
## Description

This code renders data fetched from the backend on the List component. We created a map function to map over each element in the data array and create a ListItem component to display for each element. 

Learning: We reviewed how to use Chrome React tools to look at the shape of the props being passed. 

## Related Issue

closes #1 

## Acceptance Criteria

  - [x] The streamListItems function is used in App.jsx to get the items in the current list from the Firestore database
  - [x] The list is passed into the List component as a prop named data.
  - [x] In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Screen Shot 2023-04-04 at 3 03 53 PM

### After
Capture-after-2

## Testing Steps / QA Criteria
- Open the Smart Shopping List app
- Direct browser to the list page by adding /list to the url

### CODE REVIEW

**Review by emilysellers — APPROVED** [review]
This looks great. The map function is successfully rendering each list item name.

**Review by bethmelmtv — APPROVED** [review]
All of the links. No issues with the code. Great work!

File: src/views/List.jsx [hunk 1]
```diff
@@ -7,11 +7,16 @@ export function List({ data }) {
...
-				 * of this file.
-				 */}
+				{
+					/**
+					 * TODO: write some JavaScript that renders the `data` array
+					 * using the `ListItem` component that's imported at the top
+					 * of this file.
+					 */
+					data.map((item) => (
+						<ListItem name={item.name} key={item.id} />
```
Comment (danainjax) [inline src/views/List.jsx:17]: Code looks good! You can also choose to destructure the item object to pass the attributes without dot notation, if you wish.

`{data.map(({ name, id }) => (
					<ListItem key={id} name={name} />
				))}`

**Review by danainjax — APPROVED** [review]
Great work @Amy-Pr  and @jongranados. Looks very clean.

**Pull request merged**